packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -237,123 +64,47 @@ export function buildCategoryFacet(
     throw loadReducerError;
   }
 
-  const controller = buildController(engine);
+  const coreController = buildCoreCategoryFacet(engine, props);
   const {dispatch} = engine;
-  const getState = () => engine.state;
-
-  const facetId = determineFacetId(engine, props.options);
-  const options: Required<CategoryFacetOptions> = {
-    facetSearch: {...defaultFacetSearchOptions},
-    ...defaultCategoryFacetOptions,
-    ...props.options,
-    facetId,
-  };
-
-  validateOptions(
-    engine,
-    categoryFacetOptionsSchema,
-    options,
-    'buildCategoryFacet'
-  );
-
-  const createFacetSearch = () => {
-    const {facetId, facetSearch} = options;
-    const facetSearchOptions = {facetId, ...facetSearch};
-
-    return buildCategoryFacetSearch(engine, {options: facetSearchOptions});
-  };
-
-  const getRequest = () => {
-    return categoryFacetRequestSelector(engine.state, facetId);
-  };
-
-  const getResponse = () => {
-    return categoryFacetResponseSelector(engine.state, facetId);
-  };
-
-  dispatch(registerCategoryFacet(options));
-
-  const facetSearch = createFacetSearch();
-  const {state, ...restOfFacetSearch} = facetSearch;
+  const getFacetId = () => coreController.state.facetId;
```

Comment:
This was previously using `determineFacetId`. I didn't check the code at the time, I thought it was a simple deterministic selector, but it can actually generate a new ID. So we might have had a mismatch between the facetId here VS the one in the core controller.

I *think* this is what was causing analytics issues in the FTs